front/src/components/boxs/device-in-room/device-features/SensorDeviceFeature.jsx

```diff
@@ -1,18 +1,22 @@
+/* eslint-disable preact-i18n/no-unknown-key */
 import { Text } from 'preact-i18n';
 import get from 'get-value';
 import cx from 'classnames';
 import dayjs from 'dayjs';
 import relativeTime from 'dayjs/plugin/relativeTime';
+import { DEVICE_FEATURE_CATEGORIES, DEVICE_FEATURE_TYPES } from '../../../../../../server/utils/constants';
+const { getCardinalDirection } = require('../../../../utils/cardinalPoints');
 
 dayjs.extend(relativeTime);
 
-import { DEVICE_FEATURE_CATEGORIES } from '../../../../../../server/utils/constants';
-
-const SPECIAL_SENSORS = [
+const SPECIAL_SENSORS_CATEGORY = [
   DEVICE_FEATURE_CATEGORIES.OPENING_SENSOR,
   DEVICE_FEATURE_CATEGORIES.MOTION_SENSOR,
-  DEVICE_FEATURE_CATEGORIES.PRESENCE_SENSOR
+  DEVICE_FEATURE_CATEGORIES.SWITCH,
+  DEVICE_FEATURE_CATEGORIES.INDEX
 ];
+const SPECIAL_SENSORS_TYPE = [DEVICE_FEATURE_CATEGORIES.SWITCH.BINARY];
+const SPECIAL_SENSORS_CATEGORY_TYPE = [DEVICE_FEATURE_CATEGORIES.ANGLE_SENSOR + DEVICE_FEATURE_TYPES.SENSOR.STRING];
```

Comment:
All this lists make file hard to read. It is not clear to understand if condition is verified or not.